src/core/lib/composite-layer.js

```diff
@@ -71,11 +66,19 @@ export default class CompositeLayer extends Layer {
     };
   }
 
+  // Called by layer manager to render sublayers
   _renderLayers(updateParams) {
-    if (this.state.oldSubLayers && !this.shouldUpdateState(updateParams)) {
-      log.log(2, `Composite layer reused sublayers ${this}`, this.state.oldSubLayers);
-      return this.state.oldSubLayers;
-    }
+    // TODO - won't updateLayer also be called? Avoid "double diffing"
+    // const {oldProps, props} = updateParams;
+    // this.diffProps(oldProps, props);
+
+    // TODO - Bug: Composite layers need to call the shouldUpdateState method
+    // of all its cached sublayers to ensure they all get properly updated
+    // it is not a "reuse all or none"...
```

Comment:
The problem I think I am seeing is  that we are calling `this.shouldUpdateState` rather than `subLayer.shouldUpdateState`. How can the composite layer decide if the rendered layer needs to update?

Reply:
`this.shouldUpdateState` only determines if a new set of sub layers will be generated. LayerManager will call `shoudlUpdateState` on each sublayer.

Reply:
OK I will drop this change. I am quite convinced there are very serious bugs in our composite layer handling, but makes sense to address them in a separate PR.